compiler-core/src/ast.rs

```diff
@@ -282,6 +282,41 @@ impl TypeAst {
             | TypeAst::Constructor { location, .. } => *location,
         }
     }
+
+    pub fn clear_location(&self) -> TypeAst {
```

Comment:
This is expensive, we shouldn't be making a new copy of the TypeAst, instead we should have a comparison function

Reply:
Fixed to use comparison functions